ios/ScannerBle.swift

```diff
@@ -18,50 +18,22 @@
 import CoreBluetooth
 
 ///
-/// BeaconScannerDelegate
-///
-/// Implement this to receive notifications about beacons.
-protocol BeaconScannerDelegate {
-  func urlContextChanged(beaconScanner: BeaconScanner)
-}
-
-///
 /// BeaconScanner
 ///
 /// Scans for Eddystone compliant beacons using Core Bluetooth. To receive notifications of any
 /// sighted beacons, be sure to implement BeaconScannerDelegate and set that on the scanner.
 ///
-class BeaconScanner: NSObject, CBCentralManagerDelegate {
-  
-  var delegate: BeaconScannerDelegate?
-  
+class BeaconScanner: NSObject, Scanner, CBCentralManagerDelegate {
   private var centralManager: CBCentralManager!
   private let beaconOperationsQueue: dispatch_queue_t = dispatch_queue_create("beacon_operations_queue", nil)
   private var shouldBeScanning: Bool = false
-  private var _urlFound = Set<NSURL>()
-  
-  var urlFound: Set<NSURL> {
-    get {
-      return _urlFound
-    }
-  }
-  
-  var urls: Array<String> {
-    get {
-      var urls = Set<String>();
-      for url in _urlFound {
-        urls.insert("\(url)");
-      }
-      return Array(urls);
-    }
-  }
+  private var callback: ((Dictionary<String, AnyObject>) -> Void)!
   
-  override init() {
+  init(callback: (Dictionary<String, AnyObject>) -> Void) {
```

Comment:
I liked the way that we were using before, defining a delegate so the core class needs to implements the delegate. Seems to me more ios than passing the callback function.